sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -1025,6 +1036,20 @@ class SumBinaryFixer(toType: DataType, isAnsi: Boolean)
   private var previousResult: Option[Scalar] = None
   private var previousOverflow: Option[Scalar] = None
 
+  // checkpoint
+  private var _previousResult: Option[Scalar] = None
+  private var _previousOverflow: Option[Scalar] = None
+
+  override def checkpoint(): Unit = {
+    _previousOverflow = previousOverflow
+    _previousResult = previousResult
+  }
+
+  override def restore(): Unit = {
+    previousOverflow = _previousOverflow
```

Comment:
What happens if the value is set to something that is not the same as `_previous*`? Are we going to leak that Scalar value? This goes for all of the checkpoint code.  We have to be careful that if it is the exact same scalar value we don't want to close it when we do the restore.